cql3/restrictions/statement_restrictions.cc

```diff
@@ -1033,7 +1033,7 @@ struct intersection_visitor {
 
     value_set operator()(const nonwrapping_range<bytes>& a, const nonwrapping_range<bytes>& b) const {
         const auto common_range = a.intersection(b, type->as_tri_comparator());
-        return common_range ? *common_range : empty_value_set;
+        return !common_range.empty(type->as_tri_comparator()) ? std::move(common_range) : empty_value_set;
```

Comment:
Hm, I didn't even consider it when coding this patch - it was just mechanical replacement from `std::optional` to calling `empty()`. It would be nice to simplify that in a follow-up though.

Reply:
Yeah, a follow-up is the right place to do it.

Reply:
Agree